components/camera/videosource/webcam.go

```diff
@@ -203,7 +205,29 @@ func NewWebcamSource(ctx context.Context, attrs *WebcamAttrs, logger golog.Logge
 	constraints := makeConstraints(attrs, debug, logger)
 
 	if attrs.Path != "" {
-		return tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		cam, err := tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		if err != nil {
+			return cam, err
+		}
+
+		utils2.PanicCapturingGo(func() {
+			ticker := time.NewTicker(time.Second / 2)
+			defer ticker.Stop()
+			defer utils2.TryClose(ctx, cam)
+
+			for {
+				select {
+				case <-ticker.C:
+					if _, err := getStatusVideoSource(attrs.Path, false, constraints, logger); err != nil {
```

Comment:
What's going to reopen the camera if this closes it behind other caller's backs? This goroutine also needs to respect the Close of the camera resource itself such that it waits to shut this goroutine down.

Reply:
The first question shouldn't be an issue anymore since I'm no longer using `getStatusVideoSource` to get the source. I'm not sure if the second part is still relevant though.

As far as re-opening the camera in general that'll be done in [DATA-597](https://viam.atlassian.net/browse/DATA-597).